include/eosio/vm/parser.hpp

```diff
@@ -644,20 +665,28 @@ namespace eosio { namespace vm {
             pc_stack.pop_back();
          };
 
+         auto check_in_bounds = [&]{
+            EOS_VM_ASSERT(!detail::get_allow_code_after_function_end(_options) || !pc_stack.empty(),
+                          wasm_parse_exception, "code after function end");
+         };
+
          while (code.offset() < bounds) {
             EOS_VM_ASSERT(pc_stack.size() <= detail::get_max_nested_structures(_options), wasm_parse_exception,
                           "nested structures validation failure");
 
             switch (*code++) {
-               case opcodes::unreachable: code_writer.emit_unreachable(); op_stack.start_unreachable(); break;
+               case opcodes::unreachable: check_in_bounds(); code_writer.emit_unreachable(); op_stack.start_unreachable(); break;
```

Comment:
Any reason to have these for every instruction and not just placed above the switch?

Reply:
It isn't in every opcode (unfortunately).

Reply:
Gotcha.